omnisette/src/lib.rs

```diff
@@ -27,30 +28,49 @@ pub mod remote_anisette;
 pub struct AnisetteHeaders;
 
 #[allow(dead_code)]
-#[derive(Debug)]
-enum AnisetteMetaError {
+#[derive(Debug, Error)]
+pub enum AnisetteError {
     #[allow(dead_code)]
+    #[error("Unsupported device")]
     UnsupportedDevice,
+    #[error("Invalid argument {0}")]
     InvalidArgument(String),
+    #[error("Anisette not provisioned!")]
+    AnisetteNotProvisioned,
+    #[error("Plist serialization error {0}")]
+    PlistError(#[from] plist::Error),
+    #[error("Request Error {0}")]
+    ReqwestError(#[from] reqwest::Error),
+    #[cfg(feature = "remote-anisette-v3")]
+    #[error("Provisioning socket error {0}")]
+    WsError(#[from] tokio_tungstenite::tungstenite::error::Error),
+    #[cfg(feature = "remote-anisette-v3")]
+    #[error("JSON error {0}")]
+    SerdeError(#[from] serde_json::Error),
+    #[error("IO error {0}")]
+    IOError(#[from] io::Error),
+    #[error("ADI error {0}")]
+    ADIError(#[from] ADIError),
+    #[error("Invalid library format")]
+    InvalidLibraryFormat,
+    #[error("Misc")]
+    Misc,
+    #[error("Missing Libraries")]
+    MissingLibraries,
+    #[error("{0}")]
+    Anyhow(#[from] anyhow::Error)
```

Comment:
Did you end up keeping some code using anyhow?

Reply:
Don't think so, why?

Reply:
So why are we keeping that case in the enum? It could allow us to remove the anyhow crate dep altogether couldn't it?

Reply:
![image](https://github.com/SideStore/apple-private-apis/assets/11222084/1a866371-5d4d-47b2-abf8-1f25306779fc)
nvmind we do :sob: not me tho

Reply:
I guess that'll be for another PR then